pkg/keda/scaleobject_test.go

```diff
@@ -0,0 +1,593 @@
+		client              bool
+	}{
+		{
+			name:                defaultScaledObjectName,
+			namespace:           defaultScaledObjectNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           defaultScaledObjectNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("scaledObject 'name' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                defaultScaledObjectName,
+			namespace:           "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("scaledObject 'nsname' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "triggerauthtest",
+			namespace:           defaultScaledObjectNamespace,
+			addToRuntimeObjects: false,
+			expectedError: fmt.Errorf("scaledObject object triggerauthtest does not exist " +
+				"in namespace test-appspace"),
+			client: true,
+		},
+		{
+			name:                "triggerauthtest",
+			namespace:           defaultScaledObjectNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("scaledObject 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testTriggerAuth := generateScaleObject(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testTriggerAuth)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := PullScaleObject(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		} else {
+			assert.Equal(t, testTriggerAuth.Name, builderResult.Object.Name)
+			assert.Equal(t, testTriggerAuth.Namespace, builderResult.Object.Namespace)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestNewScaledObjectBuilder(t *testing.T) {
+	testCases := []struct {
+		name          string
+		namespace     string
+		expectedError string
+		client        bool
+	}{
+		{
+			name:          defaultScaledObjectName,
+			namespace:     defaultScaledObjectNamespace,
+			expectedError: "",
+			client:        true,
+		},
+		{
+			name:          "",
+			namespace:     defaultScaledObjectNamespace,
+			expectedError: "scaledObject 'name' cannot be empty",
+			client:        true,
+		},
+		{
+			name:          defaultScaledObjectName,
+			namespace:     "",
+			expectedError: "scaledObject 'nsname' cannot be empty",
+			client:        true,
+		},
+		{
+			name:          defaultScaledObjectName,
+			namespace:     defaultScaledObjectNamespace,
+			expectedError: "",
+			client:        false,
```

Comment:
If there should be an error if the client is `false`. In this case the builder needs a check the client is being passed.

Reply:
there is no object to get an error from it